Sources/Data/Context/DeviceNameManager.swift

```diff
@@ -0,0 +1,10 @@
+//
+//  DeviceNameManager.swift
+//  
+//
+//  Created by Biso Shrestha on 2022-11-08.
+//
+
+public protocol DeviceNameManager {
+    func setDeviceName(_ deviceName: String)
+}
```

Comment:
actually I wonder that in lieu of having this separate protocol, you could just add it to StaticContextProvider. and put the persistedvalue itself on ContextManager like the others.

Or just duplicate it into RoverSettingsViewController like was done with testdevice.

Reply:
Are we going to allow this value to be set in release apps/builds?  I wasn't sure or I would have put it into the existing DebugContextProvider.

Reply:
ya works in prod builds.